blueprints/traffic.portal.objects.md

```diff
@@ -0,0 +1,562 @@
+all the details that entails. A Traffic Portal can have updates and
+revalidations pending, has a physical location, is associated with a CDN
+despite being primarily for operating arbitrary numbers of CDNs, belongs to a
+Cache Group despite not being a Cache Server, has a Type with no semantic
+meaning beyond identifying it as a Traffic Portal instance, has a Profile with
+no meaningful Parameters, has a HashID despite not being hashed for routing by
+Traffic Router, and can have multiple network interfaces with their own
+bandwidth limits - despite that Traffic Monitor does not monitor Traffic Portals
+at all.
+
+## Proposed Change
+All of these properties are superfluous with no real meaning for Traffic
+Portals, and for that reason alone they'd be better off as their own object.
+The new Traffic Portal object would be compact, containing only the properties
+that are needed by and make sense on Traffic Portal instances. Doing this would
+also be a step toward avoiding complex filtering on servers for processes that
+are looking for specific kinds of servers (though as far as this writer knows,
+there aren't any that look specifically for Traffic Portal servers).
+
+## Data Model Impact
+The proposed new object type is expressed below as a TypeScript interface.
+
+```typescript
+interface TrafficPortal {
+	ipv4Address: string | null;
+	ipv6Address: string | null;
+	notes: string;
+	online: boolean;
+	tags: Set<string>;
+	url: URL;
+}
+```
+
+Each property is further described in the following subsections.
+
+### IPv4 Address
+This string contains a valid IPv4 Address at which the UI is served. If it is
+`null`, then it is assumed that there is no static IPv4 Address at which
+the Traffic Portal instance may be reached - or it does not communicate over
+IPv4 at all.
+
+### IPv6 Address
+This string contains a valid IPv6 Address at which the UI is served. If it is
+`null`, then it is assumed that there is no static IPv6 Address at which
+the Traffic Portal instance may be reached - or it does not communicate over
+IPv6 at all.
+
+### Notes
+This string contains arbitrary text for containing miscellaneous information.
+
+### Online
+This boolean value indicates whether or not this Traffic Portal is online and
+serving its user interface.
+
+### Tags
+The Tags associated with a Traffic Portal are represented by a set of strings
+that are Tag Names.
+
+(Note that this property should only be added if the implementation is done
+after #4819 is merged - if ever.)
+
+### URL
+This string is a full URL at which the UI is served, including scheme (e.g.
+`http://`) and optionally port (e.g. `80`), e.g.
+`https://trafficportal.infra.ciab.test:443`.
+
+If the port is omitted, it will be guessed based on the protocol indicated by
+the schema, e.g. `80` for `http://`, which indicates the HTTP protocol. The only
+valid schemes are `https://` and `http://`.
+
+This property uniquely identifies a Traffic Portal, and must therefore
```

Comment:
Are you also attempting to cover the use case of a LB sitting in front of multiple TP instances?